### PR TITLE
Tracedb interface && cli

### PR DESCRIPTION
**Pull request by debris**
#### this pr should be merged after:

#992 #994 #996

### rpc docs

https://gist.github.com/debris/051c131e877affefeab4553509640a43

#### this pr includes:

- client api for tracing
- actual storing traces in database
- tracing cli
- traces rpc

### CODE REVIEW

**Comment by gavofyork** [comment]
will review after #995 merged to minimise the clutter.

looking at what is generated, there's at least one typo.

**Comment by gavofyork** [comment]
`TraceID` (an identifier substring used in previous code) versus `TraceId` in this PR. that said, i think this is endemic in the codebase and we'll need a proper sweep to get rid of this inconsistency.

File: parity/cli.rs [hunk 1]
```diff
@@ -103,6 +103,11 @@ Sealing/Mining Options:
                            be included in next block) [default: 1024].
 
 Footprint Options:
+  --tracing BOOL           Indicates if full transaction tracing should be
+                           enabled. Works only if client had been fully synced with
+                           tracing enabled. BOOL may be one of auto, on, off.
+                           auto uses last used value of this option (off it does
```
Comment (gavofyork) [inline parity/cli.rs]: "off *if* it does"

File: parity/rpc.rs [hunk 2]
```diff
@@ -115,6 +115,10 @@ pub fn setup_rpc_server(
 				// not adding to modules, since `ethcore` is not supported in geth
 				server.add_delegate(EthcoreClient::new(&deps.miner, deps.logger.clone(), deps.settings.clone()).to_delegate())
 			},
+			"traces" => {
+				// not adding to modules, since `traces` is not supported in geth
+				server.add_delegate(TracesClient::new(&deps.client).to_delegate())
```
Comment (tomusdrw) [inline parity/rpc.rs]: Should be also included in `parity/webapps.rs`?

Comment (debris) [inline parity/rpc.rs]: I have no idea... It's not a part of 'standard' rpc, but it would be useful

**Comment by debris** [comment]
> TraceID (an identifier substring used in previous code) versus TraceId in this PR. that said, i think this is endemic in the codebase and we'll need a proper sweep to get rid of this inconsistency.

What do you mean? Which `TraceID`?

File: parity/rpc.rs (hunk 2 above)
Comment (tomusdrw) [inline parity/rpc.rs]: I think it should be added just for consistency.

**Comment by gavofyork** [comment]
Was thinking of `CacheID`, sorry.

**Pull request merged**